mesonbuild/interpreter.py

```diff
@@ -2792,34 +2795,66 @@ def func_add_languages(self, node, args, kwargs):
             return False
         return self.add_languages(args, required)
 
-    def get_message_string_arg(self, node):
-        # reduce arguments again to avoid flattening posargs
-        (posargs, _) = self.reduce_arguments(node.args)
-        if len(posargs) != 1:
-            raise InvalidArguments('Expected 1 argument, got %d' % len(posargs))
-
-        arg = posargs[0]
-        if isinstance(arg, list):
-            argstr = stringifyUserArguments(arg)
-        elif isinstance(arg, dict):
-            argstr = stringifyUserArguments(arg)
-        elif isinstance(arg, str):
-            argstr = arg
-        elif isinstance(arg, int):
-            argstr = str(arg)
-        else:
-            raise InvalidArguments('Function accepts only strings, integers, lists and lists thereof.')
-
-        return argstr
+    def get_message_string_arg(self, args):
+        if len(args) != 1:
+            raise InvalidArguments('Expected 1 argument, got %d' % len(args))
```

Comment:
There's pending patches to get rid of the `%` formatter, so lets not make more work for him and use `str.format`

Reply:
I'm not changing that line, let's not make changes unrelated to this PR.